MCMC/simulation.py

```diff
@@ -197,11 +202,37 @@ def visualize(self, frame_number=0, save_path=None):
             fig_name = "system_frame_" + str(frame_number)
             plt.savefig(os.path.join(save_path, fig_name))
         return plt
-
-    def trajectory(self, save_path=""):
+    
+    def save_system(self, frame=-1):
         """
-        Find a way to play the trajectory from the system history.
-        Note: matplotlib.animation might be helpful(https://matplotlib.org/stable/api/animation_api.html)
-        :param save_path: Path to save the trajectory.
+        Save a single snapshot of system_history to a .txt file.
+        :param frame: Index number of system_history
         """
+        system_array = self.system_history[frame]
+        np.savetxt(fname="system.txt", X=system_array)
+
+    def _update_trajectory(self):
+        with gsd.hoomd.open("trajectory.gsd", "wb") as traj:
+            for sys in self.system_history:
+                snap = gsd.hoomd.Snapshot()
+                snap.particles.N = self.n_particles 
+                snap.configuration.box = np.array([self.L, self.L, self.L, 0, 0, 0])
+                snap.particles.types = ["A"]
+                coords = np.append(
+                        sys, np.zeros((sys.shape[0], 1)), axis=1
+                )
+                snap.particles.position = coords 
+                traj.append(snap)
+
+    def _update_log_file(self):
+        if not os.path.isfile("log.txt"):
+            with open("log.txt", "w") as file:
+                for e, t in zip(self.energies, self.temperatures):
+                    file.write(f"{e},{t}"+"\n")
+        else:
+            with open("log.txt", "a") as file:
+                for e, t in zip(self.energies, self.temperatures):
+                    file.write(f"{e},{t}"+"\n")
```

Comment:
What do you think about having a user-defined `log_path` in the init? In that case, parallel runs won't overwrite the log files, unless Signac has a mechanism to avoid that.

Reply:
Signac avoids all that stuff :)